fastlane/spec/actions_specs/push_to_git_remote_spec.rb

```diff
@@ -114,8 +117,71 @@
         expect(result).to eq("git push origin master:master --tags --push-option=something-to-tell-remote-git-server --push-option=something-else")
       end
 
-      context "runs git push without local_branch" do
-        it "should raise an error if get current branch failed" do
+      context "runs git push using a checked out commit git branch" do
+        it "should use the checked out commit git branch" do
+          allow(Fastlane::Actions).to receive(:sh)
+            .with("git rev-parse --abbrev-ref HEAD", log: false)
+            .and_return("staging")
+
+          result = Fastlane::FastFile.new.parse("lane :test do
+              push_to_git_remote
+            end").runner.execute(:test)
+
+          expect(result).to eq("git push origin staging:staging --tags")
+        end
+      end
+
+      context "runs git push when checked out commit git branch is different than git branch name is set in CI ENV" do
+        it "should use the checked out commit git branch if different than CI ENV" do
+          allow(Fastlane::Actions).to receive(:sh)
+            .with("git rev-parse --abbrev-ref HEAD", log: false)
+            .and_return("staging")
+          allow(Fastlane::Actions).to receive(:git_branch).and_return("master")
+
+          result = Fastlane::FastFile.new.parse("lane :test do
+              push_to_git_remote
+            end").runner.execute(:test)
+
+          expect(result).to eq("git push origin staging:staging --tags")
+        end
+      end
+
+      context "runs git push when checked out commit is detached HEAD and git branch name is set in CI ENV" do
+        it "should use CI ENV git branch if checked out commit is detached HEAD" do
+          allow(Fastlane::Actions).to receive(:sh)
+            .with("git rev-parse --abbrev-ref HEAD", log: false)
+            .and_return("HEAD")
+          allow(Fastlane::Actions).to receive(:git_branch).and_return("master")
+
+          result = Fastlane::FastFile.new.parse("lane :test do
+              push_to_git_remote
+            end").runner.execute(:test)
+
+          expect(result).to eq("git push origin master:master --tags")
+        end
+      end
+
+      context "runs git push when checked out commit is detached HEAD and git branch name is not set in CI ENV" do
+        it "should raise an error if checked out commit is detached HEAD and git branch name is not set in CI ENV" do
```

Comment:
I think this suffices:

```suggestion
        it "should raise an error" do
```

The spec reads `context runs git push when checked out commit is detached HEAD and git branch name is not set in CI ENV` `it should raise an error`